robottelo/upgrade_utility.py

```diff
@@ -0,0 +1,165 @@
+"""Common Upgrade test utilities """
+
+from wait_for import wait_for
+from fabric.api import execute, run
+from robottelo.vm import VirtualMachine, settings
+from upgrade.helpers.docker import docker_execute_command
+from robottelo.api.utils import call_entity_method_with_timeout, entities
+
+
+class CommonUpgradeUtility(object):
```

Comment:
I challenge creating this class.

1. It contains many completely unrelated methods.
2. Most of the methods could be decorated with `@staticmethod`. They are self-contained, they don't take any data from class instance (in fact, they don't care if class was instantiated at all), they don't modify instance or class.
3. The only argument it takes during instantiation is optional, and used only by three methods.

I believe this file should just contain a loose collection of functions. Then tests can import functions they need and call them directly.

Reply:
1. Yes It contains many unrelated methods but all of them are part of the upgrade, that's the reason I kept them at a single place.
2. Acknowledged.

Reply:
> 1. Yes It contains many unrelated methods but all of them are part of the upgrade, that's the reason I kept them at a single place.

And let them be in single place - `robottelo.upgrade_utility` module. But there is no point in having them in class if you are not getting any benefit out of them being in the class.

Reply:
> > 1. Yes It contains many unrelated methods but all of them are part of the upgrade, that's the reason I kept them at a single place.
> 
> And let them be in single place - `robottelo.upgrade_utility` module. But there is no point in having them in class if you are not getting any benefit out of them being in the class.

You are right Mirek,  I know there is no short term benefit of this class but in the future, if re-utilization requires then it will be useful.
what do you suggest?
As we know most of the methods are completely unrelated to each other, Can I separate them in a different file(function base/class(static) base) and make it more meaningful?

Reply:
> You are right Mirek, I know there is no short term benefit of this class but in the future, if re-utilization requires then it will be useful.

Do you have anything specific on your mind? Some scenario where it would be helpful to group these functions into a class?

Also: if that scenario requires new functions that can well work together in a class, we can just put a class into this module. There is no rule that every module should have a class, or exactly one class, or that functions can't be put into module directly etc.

> what do you suggest?

I suggest that this class is removed and all methods are stored as plain functions. Tests should import functions they need.
We can add classes later if they are beneficial.

> As we know most of the methods are completely unrelated to each other, Can I separate them in a different file(function base/class(static) base) and make it more meaningful?

Thinking about this, some of the methods that you have here probably implement patterns used in other (non-upgrade) tests, and it would make sense to move functions to `helpers` or somewhere else, and port other tests to use them. But that's probably more than you wanted to tackle in this PR, so let's just keep everything in `upgrade_utility` module for now.

Reply:
> > You are right Mirek, I know there is no short term benefit of this class but in the future, if re-utilization requires then it will be useful.
> 
> Do you have anything specific on your mind? Some scenario where it would be helpful to group these functions into a class?
> 
> Also: if that scenario requires new functions that can well work together in a class, we can just put a class into this module. There is no rule that every module should have a class, or exactly one class, or that functions can't be put into module directly etc.

Not yet Mirek, Anyway If it will come then amend it with proper discussion.

Thanks Mirek for valuable comment.

Reply:
Thanks for implementing these changes!